packages/swap/src/quoting/Leg.ts

```diff
@@ -11,22 +12,23 @@ export default class Leg {
   static of(srcAsset: ChainflipAsset, destAsset: ChainflipAsset, amount: bigint | null): Leg | null;
   static of(srcAsset: ChainflipAsset, destAsset: ChainflipAsset, amount: bigint | null) {
     if (amount === null) return null;
+    assert(srcAsset !== 'Dot' && destAsset !== 'Dot', 'Dot is not supported anymore');
     assert(srcAsset !== destAsset, 'srcAsset and destAsset must be different');
     assert(srcAsset === 'Usdc' || destAsset === 'Usdc', 'one of the assets must be Usdc');
     return new Leg(srcAsset, destAsset, amount);
   }
 
   private constructor(
-    private readonly srcAsset: ChainflipAsset,
-    private readonly destAsset: ChainflipAsset,
+    private readonly srcAsset: Exclude<ChainflipAsset, 'Dot'>,
+    private readonly destAsset: Exclude<ChainflipAsset, 'Dot'>,
     public amount: bigint,
   ) {}
 
   private getSide(): 'BUY' | 'SELL' {
     return this.destAsset !== 'Usdc' ? 'BUY' : 'SELL';
   }
 
-  getBaseAsset(): Exclude<ChainflipAsset, 'Usdc'> {
+  getBaseAsset(): Exclude<BaseChainflipAsset, 'Dot'> {
```

Comment:
I think we need to exclude both `Usdc` & `Dot` here

Reply:
baseChainflipASset excludes usdc already